object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -123,129 +119,89 @@ void checkMustCallInvoked(ControlFlowGraph cfg) {
           handleReturn((ReturnNode) node, cfg, newDefs);
         } else if (node instanceof MethodInvocationNode || node instanceof ObjectCreationNode) {
           handleInvocation(newDefs, node);
-        } else if (node instanceof TypeCastNode || node instanceof TernaryExpressionNode) {
-          handleTypeCastOrTernary(node, newDefs);
+        } else if (node instanceof TernaryExpressionNode) {
+          handleTernary(node, newDefs);
         }
       }
 
       handleSuccessorBlocks(visited, worklist, newDefs, curBlockLocals.block);
     }
   }
 
-  private void handleTypeCastOrTernary(Node node, Set<ImmutableSet<LocalVarWithTree>> defs) {
-    if (!(node instanceof TypeCastNode || node instanceof TernaryExpressionNode)) {
-      throw new BugInCF("unexpected node type " + node.getClass());
-    }
-    if (!nestedInCastOrTernary(node) && hasNestedPseudoAssignInvoke(node, defs)) {
-      checkPseudoAssignToOwning(node);
+  private void handleTernary(Node node, Set<ImmutableSet<LocalVarWithTree>> defs) {
+    LocalVariableNode ternaryLocal = typeFactory.mapTempVarToNode.inverse().get(node.getTree());
+
+    Node operand = removeCasts(((TernaryExpressionNode) node).getThenOperand());
+    LocalVariableNode operandLocal = typeFactory.mapTempVarToNode.inverse().get(operand.getTree());
+    if (operandLocal == null || !isVarInDefs(defs, operandLocal)) {
+      operand = removeCasts(((TernaryExpressionNode) node).getElseOperand());
```

Comment:
It seems like we never look at the else operand if we have a  local for the then operand.  That's not what I expected; I thought we'd handle them separately, treating each as a separate assignment to the temporary for the ternary expression.  Did I misunderstand?